public/typescript/authentication.ts

```diff
@@ -31,8 +31,9 @@ declare const publicKeyOptions: SerializedPublicKeyCredentialRequestOptions;
     updateState(S.REQUEST_USER_FOR_ASSERTION),
```

Comment:
Consider offloading the logic to some helper functions with more descriptive method names. This to lower the cyclomatic complexity.